skimage/draw/_random_shapes.py

```diff
@@ -200,13 +205,34 @@ def _generate_random_colors(num_colors, num_channels, intensity_range, random):
         each channel are drawn from the corresponding `intensity_range`.
 
     """
+
     if num_channels == 1:
-        intensity_range = (intensity_range, )
+        intensity_range_tpl = (intensity_range, )
     elif len(intensity_range) == 1:
-        intensity_range = intensity_range * num_channels
+        intensity_range_tpl = intensity_range * num_channels
+    else:
+        intensity_range_tpl = intensity_range
+
+    if np.diff(np.array(intensity_range_tpl)).sum() == 0:
+        # if intensity range only spans a single color
+        colors = np.array([np.array(intensity_range_tpl)[:, 0]])
+        msg = 'Intensity range spans only excluded intensity value.'
+        if np.array(colors == exclude).all():
+            raise ValueError(msg)
+        return np.repeat(colors, num_colors, axis=0)
+
     colors = [random.randint(r[0], r[1]+1, size=num_colors)
-              for r in intensity_range]
-    return np.transpose(colors)
+              for r in intensity_range_tpl]
+    colors = np.transpose(colors)
+
+    to_replace = np.array([(exclude == color).all() for color in colors])
```

Comment:
First convert to a numpy array and then perform array operations?

Reply:
Yes, would be more efficient.

Reply:
something like this: ```to_replace = np.equal(color, exclude).all(axis=1)```